src/main/java/org/spongepowered/api/keyboard/KeyBinding.java

```diff
@@ -96,6 +98,34 @@ static Builder builder() {
         Builder displayName(Text displayName);
 
         /**
+         * Sets the press executor of this key binding, it will be called
+         * when a specific {@link Player} the key binding pressed.
+         *
+         * @param executor The press executor
+         * @return This builder for chaining
+         */
+        Builder pressExecutor(@Nullable BiConsumer<Player, KeyBinding> executor);
+
+        /**
+         * Sets the release executor of this key binding, it will be called
+         * when a specific {@link Player} the key binding release.
+         *
+         * @param executor The release executor
+         * @return This builder for chaining
+         */
+        Builder releaseExecutor(@Nullable BiConsumer<Player, KeyBinding> executor);
+
+        /**
+         * Sets the tick executor of this key binding, it will be called
+         * when a specific {@link Player} the key binding holds the key
+         * and a tick occurred.
+         *
+         * @param executor The tick executor
+         * @return This builder for chaining
+         */
+        Builder tickExecutor(@Nullable BiConsumer<Player, KeyBinding> executor);
```

Comment:
Maybe I should make that last part more clear. I mean that you should remove `@Nullable` annotations, assume all user-provided executors are not null, and just do an internal check before firing an un-user-set executor.

Though I guess it's really potāto potáto. Three annotations won't kill me.